drivers/vulkan/rendering_device_driver_vulkan.cpp

```diff
@@ -4693,9 +4899,165 @@ void RenderingDeviceDriverVulkan::command_begin_label(CommandBufferID p_cmd_buff
+		return;
+	}
+
+	VkDeviceFaultCountsEXT fault_counts = {};
+	fault_counts.sType = VK_STRUCTURE_TYPE_DEVICE_FAULT_COUNTS_EXT;
+	VkResult vkres = device_functions.GetDeviceFaultInfoEXT(vk_device, &fault_counts, nullptr);
+
+	if (vkres != VK_SUCCESS) {
+		_err_print_error(FUNCTION_STR, __FILE__, __LINE__, "vkGetDeviceFaultInfoEXT returned " + itos(vkres) + " when getting fault count, skipping VK_EXT_device_fault report...");
+		return;
+	}
+
+	String err_msg;
+	VkDeviceFaultInfoEXT fault_info = {};
+	fault_info.sType = VK_STRUCTURE_TYPE_DEVICE_FAULT_INFO_EXT;
+	fault_info.pVendorInfos = fault_counts.vendorInfoCount
+			? (VkDeviceFaultVendorInfoEXT *)memalloc(fault_counts.vendorInfoCount * sizeof(VkDeviceFaultVendorInfoEXT))
+			: nullptr;
+	fault_info.pAddressInfos =
+			fault_counts.addressInfoCount
+			? (VkDeviceFaultAddressInfoEXT *)memalloc(fault_counts.addressInfoCount * sizeof(VkDeviceFaultAddressInfoEXT))
+			: nullptr;
+	fault_counts.vendorBinarySize = 0;
+	vkres = device_functions.GetDeviceFaultInfoEXT(vk_device, &fault_counts, &fault_info);
+	if (vkres != VK_SUCCESS) {
+		_err_print_error(FUNCTION_STR, __FILE__, __LINE__, "vkGetDeviceFaultInfoEXT returned " + itos(vkres) + " when getting fault info, skipping VK_EXT_device_fault report...");
+	} else {
+		err_msg += "** Report from VK_EXT_device_fault **\n";
+		err_msg += "Description: " + String(fault_info.description) + "\n";
+		err_msg += "Vendor infos\n";
+		for (uint32_t vd = 0; vd < fault_counts.vendorInfoCount; ++vd) {
+			const VkDeviceFaultVendorInfoEXT *vendor_info = &fault_info.pVendorInfos[vd];
+			err_msg += "Info " + itos(vd) + "\n";
+			err_msg += "   Description: " + String(vendor_info->description) + "\n";
+			err_msg += "   Fault code : " + itos(vendor_info->vendorFaultCode) + "\n";
+			err_msg += "   Fault data : " + itos(vendor_info->vendorFaultData) + "\n";
+		}
+
+		static constexpr const char *addressTypeNames[] = {
+			"NONE",
+			"READ_INVALID",
+			"WRITE_INVALID",
+			"EXECUTE_INVALID",
+			"INSTRUCTION_POINTER_UNKNOWN",
+			"INSTRUCTION_POINTER_INVALID",
+			"INSTRUCTION_POINTER_FAULT",
+		};
+		err_msg += "Addresses info\n";
+		for (uint32_t ad = 0; ad < fault_counts.addressInfoCount; ++ad) {
+			const VkDeviceFaultAddressInfoEXT *addr_info = &fault_info.pAddressInfos[ad];
+			// From https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkDeviceFaultAddressInfoEXT.html
+			const VkDeviceAddress lower = (addr_info->reportedAddress & ~(addr_info->addressPrecision - 1));
+			const VkDeviceAddress upper = (addr_info->reportedAddress | (addr_info->addressPrecision - 1));
+			err_msg += "Info " + itos(ad);
+			err_msg += "   Type            : " + String(addressTypeNames[addr_info->addressType]);
+			err_msg += "   Reported address: " + itos(addr_info->reportedAddress);
+			err_msg += "   Lower address   : " + itos(lower);
+			err_msg += "   Upper address   : " + itos(upper);
+			err_msg += "   Precision       : " + itos(addr_info->addressPrecision);
```

Comment:
```suggestion
			err_msg += "Info " + itos(ad) + "\n";
			err_msg += "   Type            : " + String(addressTypeNames[addr_info->addressType]) + "\n";
			err_msg += "   Reported address: " + itos(addr_info->reportedAddress) + "\n";
			err_msg += "   Lower address   : " + itos(lower) + "\n";
			err_msg += "   Upper address   : " + itos(upper) + "\n";
			err_msg += "   Precision       : " + itos(addr_info->addressPrecision) + "\n";
```
Following up on bruvzg's review from earlier.

Reply:
Done.

I went for putting the newline in the beginning of the hardcoded string because it has multiple advantages.

i.e.
```cpp
err_msg += "\n    Message : " + itos(lower);
```

instead of:

```cpp
err_msg += "    Message : " + itos(lower) + "\n";
```